tests/test_flask_range_voting.py

```diff
@@ -296,6 +288,44 @@ def test_vote_range_process(self):
             self.assertTrue('<h3>Next 1 elections</h3>' in output.data)
             self.assertTrue('<h3>Last 2 elections</h3>' in output.data)
 
+    def test_vote_range_revote(self):
+        """ Test the vote_range function - the re-voting part. """
+        #First we need to vote
+        self.setup_db()
+
+        user = FakeUser(['voters'], username='pingou')
+        with user_set(fedora_elections.APP, user):
+            retrieve_csrf = self.app.post('/vote/test_election3')
+            csrf_token = retrieve_csrf.data.split(
+                'name="csrf_token" type="hidden" value="')[1].split('">')[0]
+            data = {
+                '4': 1,
+                '5': 0,
+                '6': 2,
+                'action': 'submit',
+                'csrf_token': csrf_token,
+            }
+            self.app.post('/vote/test_election3', data=data, follow_redirects=True)
+        #Let's not do repetition of what is tested above we aren't testing the functionality of voting as that has already been asserted
+        #Next, we need to try revoting
+            newdata = {
+                '4': 2,
+                '5': 1,
+                '6': 1,
+                'action': 'submit',
+                'csrf_token': csrf_token,
+            }
+            output = self.app.post('/vote/test_election3', data=newdata, follow_redirects=True)
+        #Next, we need to check if the vote has been recorded
```

Comment:
Ah, I need to test the DB and check values, is this what you are eluding to :) I can write that test real fast